plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -384,24 +416,85 @@ public void setViewPrincipal(ConnectorSession session, SchemaTableName schemaVie
+        SessionContext sessionContext = convert(session);
+        Map<SchemaTableName, ConnectorViewDefinition> views = new HashMap<>();
+        for (Namespace restNamespace : namespaces) {
+            List<TableIdentifier> restViews = restSessionCatalog.listViews(sessionContext, restNamespace);
+            for (TableIdentifier restView : restViews) {
+                SchemaTableName schemaTableName = SchemaTableName.schemaTableName(restView.namespace().toString(), restView.name());
+                getView(session, schemaTableName).ifPresent(view -> views.put(schemaTableName, view));
+            }
+        }
+
+        return views;
     }
 
     @Override
     public Optional<ConnectorViewDefinition> getView(ConnectorSession session, SchemaTableName viewName)
     {
+        try {
+            View view = restSessionCatalog.loadView(convert(session), toIdentifier(viewName));
+            Optional<String> comment = Optional.ofNullable(view.properties().get(COMMENT));
+            ViewVersion currentVersion = view.currentVersion();
+            Optional<ConnectorViewDefinition> viewDefinition = Optional.empty();
+            List<ConnectorViewDefinition.ViewColumn> viewColumns = IcebergUtil.viewColumnsFromSchema(typeManager, view.schema());
+
+            for (ViewRepresentation representation : currentVersion.representations()) {
+                if (representation instanceof SQLViewRepresentation) {
+                    SQLViewRepresentation sqlView = (SQLViewRepresentation) representation;
+                    String sql = sqlView.sql();
+                    Optional<String> defaultCatalog = Optional.ofNullable(currentVersion.defaultCatalog());
+                    Optional<String> defaultNamespace = Optional.of(currentVersion.defaultNamespace().toString());
+                    if (defaultCatalog.isEmpty()) {
+                        defaultCatalog = Optional.of(catalogName.toString());
+                    }
+
+                    // Found a trino dialect view, so that should get used
+                    if (sqlView.dialect().equals(TRINO_SQL_DIALECT)) {
+                        Optional<String> owner = Optional.ofNullable(view.properties().get(TRINO_OWNER));
+                        boolean runAsInvoker = Boolean.parseBoolean(view.properties().get(TRINO_RUN_AS_INVOKER));
+                        ConnectorViewDefinition trinoViewDefinition = new ConnectorViewDefinition(
+                                sql, defaultCatalog, defaultNamespace, viewColumns, comment, owner, runAsInvoker, null);
+                        viewDefinition = Optional.of(trinoViewDefinition);
+                        break;
+                    }
+
+                    // Set a best effort view definition
+                    viewDefinition = Optional.of(new ConnectorViewDefinition(
+                            sql, defaultCatalog, defaultNamespace, viewColumns, comment, Optional.empty(), false, null));
```

Comment:
I think this will be a key point of discussion but yes best effort means "If we didn't find a Trino dialect, let's use an arbitrary SQL dialect". So here are my thoughts:

1. Most views are trivial (such as simple projections, aggregations and joins) which share the same semantics across engines. 
2. One of the benefits of having a shared view metadata is being able to access information in other dialects.
3. Considering 1 and 2, it then seems to be reasonable to simplify the user experience for Trino users, to be able to create views in other systems and then query those same views in Trino without having to do any recreation.

Now, in the current implementation this best effort behavior happens by default. To avoid nasty correctness issues what we can do is make this behavior opt in via a connector property (defaulting it to false to avoid the best effort behavior). That way users who know their domain of queries can take advantage of this benefit of Iceberg view metadata.

On Hive views, those seem to leverage Coral, which I don't think we want to do here. Instead of an IR layer that can be represented as a separate dialect in the future in Iceberg views

Reply:
Even simple projections may have different semantics.
For example Spark's `SELECT some_array[1] AS second_element ...` will incorrect return the first element in ANSI SQL, and in Trino.

AFAIK, Spark, Hive and Trino don't even agree on table name quoting  (`"` vs `` ` ``).

in Hive we have two modes

1. Coral-based, which embodies best possible approach: parse the original view text using view creator's dialect, transform the IR as necessary, output the Trino view text
2. the legacy one (`hive.hive-views.legacy-translation`) where view is transformed with a bunch of regular expressions.

If i read correctly, what is being proposed here is "the legacy mode". It feels unnatural to me to add a legacy mode where we already know how limited it is, and how it leads to correctness problems.

I like the Iceberg views IR idea. I actually advocated for it when Iceberg views were being introduced.
I understand this will be in Iceberg views v2. For now, we don't have it, and we cannot assume it will be delivered -- it's done when it's done. Working with what we have today, Coral-based (or equivalent) seems the best way to move forward.

Other than that, we could have an allow-list of patterns that we "manually" translate. This could unblock "most of those simple views" without opening a worm can. Think of it as a list of hard-coded voice patterns that take precedence over your home assistant "AI".